app/models/notification.rb

```diff
@@ -60,28 +62,9 @@ def send_new_comment_notifications_without_delay(comment)
       Notifications::NewCommentJob.perform_now(comment.id)
     end
 
-    def send_new_badge_notification(badge_achievement)
-      json_data = {
-        user: user_data(badge_achievement.user),
-        badge_achievement: {
-          badge_id: badge_achievement.badge_id,
-          rewarding_context_message: badge_achievement.rewarding_context_message,
-          badge: {
-            title: badge_achievement.badge.title,
-            description: badge_achievement.badge.description,
-            badge_image_url: badge_achievement.badge.badge_image_url
-          }
-        }
-      }
-      Notification.create(
-        user_id: badge_achievement.user.id,
-        notifiable_id: badge_achievement.id,
-        notifiable_type: "BadgeAchievement",
-        action: nil,
-        json_data: json_data,
-      )
+    def send_new_badge_achievement_notification(badge_achievement)
```

Comment:
At the deploy time, there could be old `send_new_badge_notification` jobs left. They execute `send_new_badge_notification_without_delay`, so we need to keep this method for a while. I used to be created by `handle_asynchronously`, but now we have to add it manually. This method can be removed later.

Reply:
Yes! Definitely

Reply:
I've added back both `send_new_badge_notification` and `send_new_badge_notification_without_delay` and test them